src/test/java/org/springframework/data/elasticsearch/core/SearchTemplateIntegrationTests.java

```diff
@@ -63,83 +65,264 @@ public abstract class SearchTemplateIntegrationTests {
 				"size": 100
 			  }
 			""";
-	private Script script = Script.builder() //
-			.withId("testScript") //
+
+	private static final String SEARCH_LASTNAME = """
+			{
+				"query": {
+					"bool": {
+						"must": [
+							{
+								"match": {
+									"lastName": "{{lastName}}"
+								}
+							}
+						]
+					}
+				},
+				"from": 0,
+				"size": 100
+			  }
+			""";
+
+	private static final Script SCRIPT_SEARCH_FIRSTNAME = Script.builder() //
+			.withId("searchFirstName") //
+			.withLanguage("mustache") //
+			.withSource(SEARCH_FIRSTNAME) //
+			.build();
+
+	private static final Script SCRIPT_SEARCH_LASTNAME = Script.builder() //
+			.withId("searchLastName") //
 			.withLanguage("mustache") //
-			.withSource(SCRIPT) //
+			.withSource(SEARCH_LASTNAME) //
 			.build();
 
 	@Autowired ElasticsearchOperations operations;
 	@Autowired IndexNameProvider indexNameProvider;
-	@Nullable IndexOperations indexOperations;
+	IndexOperations personIndexOperations, studentIndexOperations;
 
 	@BeforeEach
 	void setUp() {
 		indexNameProvider.increment();
-		indexOperations = operations.indexOps(Person.class);
-		indexOperations.createWithMapping();
+		personIndexOperations = operations.indexOps(Person.class);
+		personIndexOperations.createWithMapping();
+		studentIndexOperations = operations.indexOps(Student.class);
+		studentIndexOperations.createWithMapping();
+
+		operations.save( //
+				new Person("1", "John", "Smith"), //
+				new Person("2", "Willy", "Smith"), //
+				new Person("3", "John", "Myers"));
+
+		operations.save(
+				new Student("1", "Joey", "Dunlop"), //
+				new Student("2", "Michael", "Dunlop"));
 	}
 
-	@Test
-	@Order(Integer.MAX_VALUE)
-	void cleanup() {
-		operations.indexOps(IndexCoordinates.of(indexNameProvider.getPrefix() + '*')).delete();
+	@AfterEach
+	void tearDown() {
+		personIndexOperations.delete();
+		studentIndexOperations.delete();
 	}
```

Comment:
Please keep the `indexNameProvider` for the index names that cleans up at the end of the test, deleting indexes after each test is slow. I'll provide the relevant changes below